Themes/default/scripts/register.js

```diff
@@ -234,35 +189,20 @@ function smfRegister(formID, passwordDifficultyLevel, regTextStrings)
 	// Callback for getting the username data.
 	function checkUsernameCallback(XMLDoc)
 	{
-		if (XMLDoc.getElementsByTagName("username"))
-			isValid = XMLDoc.getElementsByTagName("username")[0].getAttribute("valid");
-		else
-			isValid = true;
-
-		// What to alt?
-		var alt = textStrings[isValid == 1 ? 'username_valid' : 'username_invalid'] ? textStrings[isValid == 1 ? 'username_valid' : 'username_invalid'] : '';
+		var
+			tags = XMLDoc.getElementsByTagName("username"),
+			isValid = tags && tags[0].getAttribute("valid") == 1,
+			alt = textStrings[isValid ? 'username_valid' : 'username_invalid'];
 
-		verificationFields['username'][1].className = verificationFields['username'][5] + ' ' + (isValid == 1 ? 'valid_input' : 'invalid_input');
-		setVerificationImage(verificationFields['username'][0], isValid == 1, alt);
-
-		ajax_indicator(false);
+		setVerificationImage(verificationFields['username'][2], isValid, alt);
 	}
 
 	// Set the image to be the correct type.
-	function setVerificationImage(fieldID, imageIcon, alt)
+	function setVerificationImage(el, isValid, alt)
 	{
-		if (!fieldID)
-			return false;
-		if (!alt)
-			alt = '*';
-
-		$('#' + fieldID + '_img').removeClass('valid check invalid').attr('alt', alt).attr('title', alt);
-		if (imageIcon)
-			$('#' + fieldID + '_img').addClass(imageIcon == 'check' ? 'check' : 'valid');
-		else
-			$('#' + fieldID + '_img').addClass('invalid');
-
-		return true;
+		el.previousSibling.className = isValid && isValid !== 'check' ? 'valid' : 'invalid';
+		el.textContent = isValid ? (isValid == 'check' ? '⚠️' : '✔') : '❌';
```

Comment:
Should we embedding the those icons like that.  Or should they be stored and referenced with their UTF-8 values?  I.e. store as `\x26A0`

Reply:
Doesn't really matter. The file is encoded with UTF-8, and the output is UTF-8, so it's file. They're just characters, after all.

Reply:
all js strings become utf-16 when processed in the browser

Reply:
Derp. Of course, you're right. But still, this is fine.